addons/edi_notifier/models/email_notifier.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime, timedelta
+
+from odoo import fields, api, models
+
+
+class EdiEmailNotifier(models.AbstractModel):
+
+    _name = "edi.notifier.email"
+    _inherit = "edi.notifier.model"
+    _description = "Edi Notifier Email Base Model"
+
+    def _should_notify(self, notifier, rec):
+        if rec._name != notifier.template_id.model_id.model:
+            return False
+        return True
+
+    @api.multi
+    def _notify(self, notifier, recs):
+        for rec in recs:
+            notifier.template_id.send_mail(rec.id, force_send=True)
+
+    @api.multi
+    def notify(self, notifier, recs):
+        """Filter records and send them for notification"""
+        if not recs:
+            recs = notifier.doc_type_ids
+        self._notify(notifier, self.filter_records(notifier, recs))
+
+
+class EdiEmailSuccessNotifier(models.AbstractModel):
+
+    _name = "edi.notifier.email.success"
+    _inherit = "edi.notifier.email"
+
+    def _should_notify(self, notifier, rec):
+        if rec.state != "done":
+            return False
+        return super()._should_notify(notifier, rec)
```

Comment:
Do we want to do this the other way around? first check if the model has notifier and then check the condition of this type of notifier?